spec/requests/api/v1/registrations/upload_spec.rb

```diff
@@ -0,0 +1,142 @@
+        {
+          user: {
+            full_name: 'New Name'
+          }
+        }
+      end
+
+      it 'returns the success message' do
+        subject
+        expect(json).to include_json(
+          user: {
+            id: user.id,
+            full_name: params[:user][:full_name],
+            gender: user.gender,
+            email: user.uid,
+            avatar: polymorphic_url(user.avatar)
+          }
+        )
+      end
+
+      it 'returns a success status' do
+        subject
+        expect(response).to be_successful
+      end
+    end
+
+    context 'when update personal info and ulpload perfil picture' do
+      subject { patch user_registration_url, params: params, headers: header_param }
+      let(:params) do
+        {
+          user: {
+            gender: 'female',
+            avatar: Rack::Test::UploadedFile.new('spec/fixtures/other_image.jpeg', 'image/jpeg')
+          }
+        }
+      end
+
+      it 'returns the user info uploaded' do
+        subject
+        expect(json).to include_json(
+          user: {
+            id: user.id,
+            full_name: user.full_name,
+            gender: params[:user][:gender],
+            email: user.uid,
+            avatar: polymorphic_url(user.reload.avatar)
+          }
+        )
+      end
+
+      it 'returns a success status' do
+        subject
+        expect(response).to be_successful
+      end
+    end
+
+    context 'when update perfile picture info' do
+      subject { patch user_registration_url, params: params, headers: header_param }
+
+      let(:params) do
+        {
+          user: {
+            avatar: Rack::Test::UploadedFile.new('spec/fixtures/other_image.jpeg', 'image/jpeg')
+          }
+        }
+      end
+
+      it 'returns the success message' do
+        subject
+        expect(json).to include_json(
+          user: {
+            id: user.id,
+            gender: user.gender,
+            email: user.uid,
+            avatar: polymorphic_url(user.reload.avatar)
+          }
+        )
+      end
+
+      it 'returns a success status' do
+        subject
+        expect(response).to be_successful
+      end
+    end
+  end
+
+  context 'when the request is not valid' do
+    subject { patch user_registration_url, params: params, headers: headers, as: :json }
+    context 'when the headers are' do
+      let(:params) do
+        {
+          user: {
+            full_name: 'New Name'
+          }
+        }
+      end
+
+      let(:headers) { nil }
+
+      it 'returns the error messages as a json' do
+        subject
+        message = I18n.t('devise_token_auth.registrations.user_not_found')
+        expect(json).to include_json(error: message)
+      end
+
+      it 'returns a unauthorized status' do
+        subject
+        expect(response).to have_http_status(:not_found)
+      end
+    end
+
+    context 'when params is not permited' do
+      let(:params) do
+        {
+          user: {
+            hair_color: 'Black'
```

Comment:
Maybe use a different key for this hash. Because in the future `hair_color` may be added to the model. Why don't you try using `unpermitted_param: 'value'`.